lib/mastodon/cli_helper.rb

```diff
@@ -7,3 +7,48 @@
 ActiveJob::Base.logger       = dev_null
 HttpLog.configuration.logger = dev_null
 Paperclip.options[:log]      = false
+
+module Mastodon
+  module CLIHelper
+    def create_progress_bar(total = nil)
+      ProgressBar.create(total: total, format: '%c/%u |%b%i| %e')
+    end
+
+    def parallelize_with_progress(scope)
+      ActiveRecord::Base.configurations[Rails.env]['pool'] = options[:concurrency]
+
+      progress  = create_progress_bar(scope.count)
+      pool      = Concurrent::FixedThreadPool.new(options[:concurrency])
+      futures   = []
+      aggregate = Concurrent::AtomicFixnum.new(0)
+
+      scope.reorder(nil).find_each do |item|
+        progress.total = futures.size + 1 if progress.total < futures.size + 1
+
+        futures << Concurrent::Future.execute(executor: pool) do
+          ActiveRecord::Base.connection_pool.with_connection do
+            begin
+              progress.log("Processing #{item.id}") if options[:verbose]
+
+              result = yield(item)
+              aggregate.increment(result) if result.is_a?(Integer)
+            rescue => e
+              progress.log pastel.red("Error processing #{item.id}: #{e}")
+            ensure
+              progress.increment
```

Comment:
Is `ProgressBar` thread-safe?

Reply:
I cannot find any information on this including any complaints on their bug tracker so I assume it's not going to be a problem.